bigbluebutton-html5/imports/startup/client/base.jsx

```diff
@@ -219,15 +219,31 @@ const BaseContainer = withTracker(() => {
   let breakoutRoomSubscriptionHandler;
   let meetingModeratorSubscriptionHandler;
 
-  const User = Users.findOne({ intId: credentials.requesterUserId });
-  const meeting = Meetings.findOne({ meetingId });
+  const options = {
+    approved: 1,
+    authed: 1,
+    ejected: 1,
+    color: 1,
+    effectiveConnectionType: 1,
+    extId: 1,
+    guest: 1,
+    intId: 1,
+    locked: 1,
+    loggedOut: 1,
+    meetingId: 1,
+    userId: 1,
+  };
+  const User = Users.findOne({ intId: credentials.requesterUserId }, { fields: options });
+  const meeting = Meetings.findOne({ meetingId }, { fields: {
+    meetingEnded: 1,
+    }});
   if (meeting) {
     const { meetingEnded } = meeting;
     if (meetingEnded) Session.set('codeError', '410');
   }
 
-  const approved = !!Users.findOne({ userId: Auth.userID, approved: true, guest: true });
-  const ejected = Users.findOne({ userId: Auth.userID, ejected: true });
+  const approved = !!Users.findOne({ userId: Auth.userID, approved: true, guest: true }, { fields: options });
```

Comment:
Couldn't these approved and ejected values be found in the User object that was already retrieved? Seems like a waste to request the same record multiple times.